### PR TITLE
Documentation updates and cleanup

### LINKED ISSUE #293: Update installation instructions in wiki
Since #282, installing the Hector R package can be done entirely from within R (via `install.packages`, `devtools::install_github`, or the RStudio buttons). The wiki instructions for installing Hector need to be simplified to reflect this fact.

Per my conversation with @ssmithClimate.

### PR DESCRIPTION
**Pull request by ashiklom**
- Add installation instructions for the Hector R package (#293).
- Remove references to GSL from the installation docs
- Update README to point at the new pkgdown website (PS. Perhaps someone with JGCRI admin privileges can add the "jgcri.github.io/hector" link to the GitHub repo description?)
- Group Hector functions into loosely-related groups in the function reference
- Quote paths in `makefile.standalone` to allow directories with spaces

Closes #293.

### CODE REVIEW

**Comment by ashiklom** [comment]
To facilitate the review, here is a link to the rendered version of this PR:

https://ashiklom.github.io/hector/

**Comment by ssmithClimate** [comment]
@ashiklom Perhaps in the R interface portion when running the core is mentioned (or later) say something about how the default operation re-runs from initialization forward (e.g. I assume that, if a variable is changed, everything re-runs including spin-up - which is generally what we want).

**Comment by ssmithClimate** [comment]
@ashiklom very nice! I had a few minor comments suggesting clarifications.

**Comment by ashiklom** [comment]
@ssmithClimate Thanks! I addressed your comment about `reset` in https://github.com/JGCRI/hector/pull/310/commits/33abfd09c2481aac8aee040b69a8072b13400fc1. Let me know how that looks, and if there's anything else you'd like me to change.

File: README.md [hunk 1]
```diff
@@ -4,19 +4,20 @@
...
+The primary source of Hector model documentation is the [online manual](https://jgcri.github.io/hector/articles/manual), which is included in the repository in the `vignettes/manual` directory.
+The code is also well documented with [Doxygen](http://doxygen.org)-style comments. A formal model description paper ([Hartin et al. 2015](http://www.geosci-model-dev.net/8/939/2015/gmd-8-939-2015.html)) documents its science internals and performance relative to observed data, the [CMIP5](http://cmip-pcmdi.llnl.gov/cmip5/) archive, and the reduced-complexity [MAGICC](http://www.magicc.org) model. In addition, we have developed two package vignettes demonstrating the [basics of the Hector R interface](http://jgcri.github.io/hector/articles/intro-to-hector.html) and an example application of [solving for an emissions pathway](http://jgcri.github.io/hector/articles/hector_apply.html).
```
[the remark this answers is not in the document]
Comment (ashiklom) [reply README.md]: Addressed in 3d088dc.

File: README.md [hunk 2]
```diff
@@ -4,19 +4,20 @@
...
 
 ![](https://github.com/JGCRI/hector/wiki/rcp85.png)
 
-The primary source of Hector model documentation is the Github [wiki](https://github.com/JGCRI/hector/wiki). Please note that the wiki documents are included in the repository as a git [submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules). To clone the repository as well as the documentation, you can use `git clone --recursive`. If you have already cloned non-recursively, you can fetch (download) the documentation by running `git submodule init; git submodule update`. The code is also well documented with [Doxygen](http://doxygen.org)-style comments. A formal model description paper ([Hartin et al. 2015](http://www.geosci-model-dev.net/8/939/2015/gmd-8-939-2015.html)) documents its science internals and performance relative to observed data, the [CMIP5](http://cmip-pcmdi.llnl.gov/cmip5/) archive, and the reduced-complexity [MAGICC](http://www.magicc.org) model.
+The primary source of Hector model documentation is the [online manual](https://jgcri.github.io/hector/articles/manual), which is included in the repository in the `vignettes/manual` directory.
```
[the remark this answers is not in the document]
Comment (ashiklom) [reply README.md]: Addressed in 3d088dc.

**Comment by ashiklom** [comment]
@ssmithClimate I tried to address your comments in 3d088dc. Let me know if that's sufficient.

**Comment by ssmithClimate** [comment]
@ashiklom changes look good to me!

**Pull request merged**